event_sale_pro/models/quotation_generator.py

```diff
@@ -0,0 +1,114 @@
+# -*- coding: utf-8 -*-
+# © 2015 Grupo ESOC
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import _, api, fields, models
+from .. import exceptions
+
+
+class EventQuotationGenerator(models.TransientModel):
+    _name = "event_sale_pro.quotation_generator"
+
+    event_id = fields.Many2one(
+        "event.event",
+        "Event",
+        readonly=True,
+        default=lambda self: (self.env["event.registration"]
+                              .browse(self.env.context.get("active_id"))
+                              .event_id))
+    event_type_id = fields.Many2one(
+        string="Event type",
+        readonly=True,
+        related="event_id.type")
+    product_id = fields.Many2one(
+        "product.product",
+        "Product",
+        domain="""[("sale_ok", "=", True),
+                   ("event_ok", "=", True),
+                   ("event_type_id", "=", event_type_id)]""",
+        help="Event subscription product that will be sold.")
+    registration_ids = fields.Many2many(
+        "event.registration",
+        string="Registrations",
+        relation="event_sale_pro_quotation_generator_registration_ids_rel",
+        default=lambda self: self.env.context.get("active_ids"))
+    group_by_commercial_entity = fields.Boolean(
+        default=True,
+        help="Check this box if you want to generate one quotation per "
+             "commercial entity. Leave it unchecked to generate one per "
+             "registration.")
+
+    @api.multi
+    def action_generate(self):
+        """Generate sale orders for selected event registrations."""
+        self.ensure_one()
+
+        # Check that we are only working with one event
+        if len(self.registration_ids.mapped("event_id")) > 1:
```

Comment:
Are you sure this works as intended? AFAICS two registration on the same event will still result is a list with len 2.
Maybe you want a `set` inside the `len`.